Content.Shared/SimpleStation14/Species/Shadekin/Systems/ShadekinSystem.Power.cs

```diff
@@ -0,0 +1,292 @@
+using Content.Shared.Alert;
+using Content.Shared.Rounding;
+using Content.Shared.SimpleStation14.Species.Shadekin.Components;
+using Content.Shared.SimpleStation14.Species.Shadekin.Events;
+using System.Threading.Tasks;
+
+namespace Content.Shared.SimpleStation14.Species.Shadekin.Systems
+{
+    public class ShadekinPowerSystem : EntitySystem
+    {
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly AlertsSystem _alertsSystem = default!;
+
+
+        /// <param name="PowerLevel">The current power level.</param>
+        /// <returns>The name of the power level.</returns>
+        public string GetLevelName(float PowerLevel)
+        {
+            // Placeholders
+            var powerType = Loc.GetString("shadekin-power-okay");
+            ShadekinPowerThreshold result = ShadekinPowerThreshold.Min;
+            var value = ShadekinComponent.PowerThresholds[ShadekinPowerThreshold.Max];
+
+            // Find the highest threshold that is lower than the current power level
+            foreach (var threshold in ShadekinComponent.PowerThresholds)
+            {
+                if (threshold.Value <= value && threshold.Value >= PowerLevel)
+                {
+                    result = threshold.Key;
+                    value = threshold.Value;
+                }
+            }
+
+            // Get the name of the threshold
+            switch (result)
+            {
+                case ShadekinPowerThreshold.Max:
+                    powerType = Loc.GetString("shadekin-power-max");
+                    break;
+                case ShadekinPowerThreshold.Great:
+                    powerType = Loc.GetString("shadekin-power-great");
+                    break;
+                case ShadekinPowerThreshold.Good:
+                    powerType = Loc.GetString("shadekin-power-good");
+                    break;
+                case ShadekinPowerThreshold.Okay:
+                    powerType = Loc.GetString("shadekin-power-okay");
+                    break;
+                case ShadekinPowerThreshold.Tired:
+                    powerType = Loc.GetString("shadekin-power-tired");
+                    break;
+                case ShadekinPowerThreshold.Min:
+                    powerType = Loc.GetString("shadekin-power-min");
+                    break;
+
+                default:
+                    powerType = Loc.GetString("shadekin-power-okay");
+                    break;
+            }
```

Comment:
I believe it'd make far more sense to replace this section with a dictionary.

```suggestion
            // Get the name of the threshold
            var powerDictionary = new Dictionary<ShadekinPowerThreshold, string>
            {
                {ShadekinPowerThreshold.Max, Loc.GetString("shadekin-power-max")},
                {ShadekinPowerThreshold.Great, Loc.GetString("shadekin-power-great")},
                {ShadekinPowerThreshold.Good, Loc.GetString("shadekin-power-good")},
                {ShadekinPowerThreshold.Okay, Loc.GetString("shadekin-power-okay")},
                {ShadekinPowerThreshold.Tired, Loc.GetString("shadekin-power-tired")},
                {ShadekinPowerThreshold.Min, Loc.GetString("shadekin-power-min")}
            };
```

Using this line to get the value
`powerType = powerDictionary.TryGetValue(result, out var value) ? value : Loc.GetString("shadekin-power-okay");`